hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/TransactionModule.java

```diff
@@ -90,7 +90,10 @@ static HederaEvmContext provideHederaEvmContext(
     @TransactionScope
     static Supplier<HederaWorldUpdater> provideFeesOnlyUpdater(
             @NonNull final HederaWorldUpdater.Enhancement enhancement, @NonNull final EvmFrameStateFactory factory) {
-        return () -> new ProxyWorldUpdater(enhancement, requireNonNull(factory), null);
+        return () -> {
+            enhancement.operations().begin();
```

Comment:
Is this to start anew after the `updater.revert` at `TransactionProcessor` L155 below?  

If so, then this is a "matched set" that is unfortunately widely separated in the source code.  And also _all_ new feesOnlyUpdaters will start a child operation - is that intended? 

Anyway it would be nicer IMO that _if_ it is a matched set and _if_ this is the only place it happens (I mean over there in `TransactionProcessor` that the `updater.revert` and `operations.begin` were closer together in the source code in some fashion.  (perhaps wrapped up in a method together)

Reply:
Hmm yes you're right, this is `begin()` will only be correct we have already `revert()`-ed the `RootProxyWorldUpdater` whose `commit()` threw a `ResourceExhaustedException`. 

Will ponder how we might reorganize this. 🤔